src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java

```diff
@@ -55,57 +58,30 @@ public boolean isApplicable(Class<? extends Job> jobType) {
 
         @Override
         public void onFinalized(Run run) {
-            File log = run.getLogFile();
-            if (log.getName().endsWith(".gz")) {
-                // ignore already compressed log
-                LOGGER.log(Level.FINER, String.format("Skipping %s because the log is already compressed", run));
-                return;
-            }
 
             if (run.getParent().getProperty(BuildLogCompressor.class) == null) {
                 LOGGER.log(Level.FINER, String.format("Skipping %s because the project is not configured to have compressed logs", run));
                 return;
             }
 
-            String gzippedLogName = log.getName() + ".gz";
-
-            if (log.getName().equals("log")) {
-                LOGGER.log(Level.FINE, String.format("Compressing build log of %s", run));
-
-                // regular, expected log file
-                FileInputStream fis = null;
-                FileOutputStream fos = null;
-                GZIPOutputStream gzos = null;
-
-                try {
-                    fis = new FileInputStream(log);
-                    fos = new FileOutputStream(new File(log.getParentFile(), gzippedLogName));
-                    gzos = new GZIPOutputStream(fos);
-                    int copiedBytes = IOUtils.copy(fis, gzos);
-
-                    if (copiedBytes != log.length()) {
-                        LOGGER.log(Level.WARNING, String.format("Expected to copy %d bytes but copied %d from %s", copiedBytes, log.length(), log.getAbsolutePath()));
-                    }
-
-                    gzos.finish();
-                    LOGGER.log(Level.FINE, String.format("Finished compressing build log of %s", run));
-                } catch (IOException e) {
-                    LOGGER.log(Level.WARNING, String.format("Failed to compress build log of %s to %s", run, gzippedLogName));
-                    return;
-                } finally {
-                    IOUtils.closeQuietly(fis);
-                    IOUtils.closeQuietly(fos);
-                    IOUtils.closeQuietly(gzos);
-                }
-
-                // XXX try multiple times because Windows?
-                if (!log.delete()) {
-                    LOGGER.log(Level.WARNING, String.format("Failed to delete build log of %s after compression", run));
-                }
+            Util.compressLogFile(run);
+
+            Plugin mvn = Jenkins.getInstance().getPlugin("maven-plugin");
+            if (mvn!= null) {
+                // compress module logs, if this is a maven reactor build
+                Class<?> c;
+				try {
+					c = Class.forName("org.jenkinsci.plugins.compressbuildlog.MavenModuleLogCompressor");
```

Comment:
Sorry, I dont' get it. Could you be more specific? Check against what exactly? No matter, if this is a MavenBuild it should compress the log.